### ISSUE TITLE
Setup testing

### ISSUE THREAD

**Issue by cedricium** [cedricium/should-i-cli#8]
This is my first repo that has garnered some popularity - this would be a perfect candidate to setup some testing so others wanting to get familiar with the project and / or contribute can see how it works, and so we can cover our 🍑 to ensure regressions are introduced as more features are added.

**Comment by cedricium** [comment]
I think Mocha + Chai is definitely the way to go, but after spending several hours not getting a single test setup correctly, I'm gonna need some help with this. If anyone wants to open a PR with some simple tests, I would greatly appreciate it 😄

**Comment by andersonjoseph** [comment]
@cedricium Hi! I have some experience using Mocha + Chai for unit testing, I could do it.

**Comment by cedricium** [comment]
@andersonjoseph that'd be awesome! Please feel free to open another PR, would love to see how it's done. @oclif (used as the base CLI for `should-i-cli`) has their own test helpers, you might want to check that out: https://github.com/oclif/test

**Comment by andersonjoseph** [comment]
@cedricium hey! I couldn't play with oclif test helpers, I had the error 'command not found' over and over again. I tried to do it manually but no luck. Even so, I wrote some unit tests that works on 'utils' functions.
The file tests looks like this:

```javascript

const expect = require('chai').expect;

//testing utils
const {generateYesOrNo} = require('../src/utils/index.js');
describe('Testing answers', () => {
  it('should return yes or no', () => {
    expect(generateYesOrNo('Should I study now?')).to.match(/(yes|no)/);
    expect(generateYesOrNo('Should I sleep early?')).to.match(/(yes|no)/);
    expect(generateYesOrNo('Should I test my code before production?'))
          .to.match(/(yes|no)/);
  });
});

describe('Testing forbidden questions', () => {
  it('should return no', () => {
    expect(generateYesOrNo('Should I kill myself?')).to.equal('no');
    expect(generateYesOrNo('Should I rob a car?')).to.equal('no');
    expect(generateYesOrNo('Should I take my own life?')).to.equal('no');
  });
});

const {isAQuestion} = require('../src/utils/index.js');
describe('Testing if arg is a question', () => {
  it('should return true for questions', () => {
    expect(isAQuestion('Should I study now?')).to.be.true;
    expect(isAQuestion('Should I sleep early?')).to.be.true;
    expect(isAQuestion('Should I test my code before production?')).to.be.true;
  });

  it('should return false for no-questions', () => {
    expect(isAQuestion('Should I study now')).to.be.false;
    expect(isAQuestion('hello boys')).to.be.false;
    expect(isAQuestion('Im not a question')).to.be.false;
  });
});
```

What do you think?

**Comment by cedricium** [comment]
@andersonjoseph this is awesome, nice work! One suggestion:

In the last `it()`, maybe change `expect(isAQuestion('Im not a question')).to.be.false;` to test only numbers and/or symbols, like:

```
it('should return false for no-questions', () => {
  expect(isAQuestion('32254-0';./.;')).to.be.false;
  //                  ^^^^^^^^^^^^ number and symbols should also be classified as not a question
});
```

You introduced a regex to only match letters in your last PR and the above `expect()` should pass according to that. With that I'd say definitely open another PR 🎉

**Comment by cedricium** [comment]
Closed via 70f989fd9ee444203dd7b15f98bbd338b6b99979
